### DISCUSSION TITLE
Shader madness!

### DISCUSSION THREAD

**Discussion by KartalToth240560** [bevyengine/bevy#15431] [Q&A]
Hi!

I am trying to create a visualization of a 3D array using only just points - simple pixels, not spheres. A point for each element in the array where the value is greater than 0. I am currently at doing the custom pipeline item example.
However I now know how to send a buffer to the GPU (16^3 bytes) and that is it basically. I do not know if I can get the index of which element the shader is currently processing, because if I could I can calculate the 3D point. I also do not know why I cannot access the camera matrix in the wgsl shader. I cannot project the object space position, they show up as display positions. I have so many questions, and I have been doing my research. I just started uni and it takes up so much time, I cannot start my journey. I think it is not a hard project, it is just a very new topic for me, and some push would be much appreciated!

**Comment by 68317fa2** [comment]
Could you please give a little more in-depth view about what you are trying to achieve? Do you want to render voxels? Why can you not just take very small spheres? That would definitely be performant enough, if you only have 1024 points.
If you want to render voxels, you would be best, if you would write a custom raytracer, but that isn't easy, especially if you don't have a lot of knowledge in bevy's rendering system.
